packages/frontend/src/Build.spec.ts

```diff
@@ -0,0 +1,136 @@
+    tag: 'latest',
+    type: 'iso',
+    folder: '/tmp/image1',
+    arch: 'x86_64',
+  },
+];
+
+// Mocked bootc images, with one containing the 'bootc' and 'containers.bootc' labels, and the other not
+const mockBootcImages: ImageInfo[] = [
+  {
+    Id: 'image1',
+    RepoTags: ['image1:latest'],
+    Labels: {
+      bootc: 'true',
+    },
+    engineId: 'engine1',
+    engineName: 'engine1',
+    ParentId: 'parent1',
+    Created: 0,
+    VirtualSize: 0,
+    Size: 0,
+    Containers: 0,
+    SharedSize: 0,
+  },
+  {
+    Id: 'image2',
+    RepoTags: ['image2:latest'],
+    Labels: {
+      bootc: 'true',
+    },
+    engineId: 'engine2',
+    engineName: 'engine2',
+    ParentId: 'parent2',
+    Created: 0,
+    VirtualSize: 0,
+    Size: 0,
+    Containers: 0,
+    SharedSize: 0,
+  },
+];
+
+vi.mock('./api/client', async () => {
+  return {
+    bootcClient: {
+      listHistoryInfo: vi.fn(),
+      listBootcImages: vi.fn(),
+    },
+    rpcBrowser: {
+      subscribe: () => {
+        return {
+          unsubscribe: () => {},
+        };
+      },
+    },
+  };
+});
+
+async function waitRender(customProperties: object): Promise<void> {
+  const result = render(Build, { ...customProperties });
+  // wait that result.component.$$.ctx[2] is set
+  while (result.component.$$.ctx[2] === undefined) {
+    await new Promise(resolve => setTimeout(resolve, 100));
+  }
+}
+
+test('Render shows correct images and history', async () => {
+  vi.mocked(bootcClient.listHistoryInfo).mockResolvedValue(mockHistoryInfo);
+  vi.mocked(bootcClient.listBootcImages).mockResolvedValue(mockBootcImages);
+  await waitRender(Build);
+
+  // Wait until children length is 2 meaning it's fully rendered / propagated the changes
+  while (screen.getByLabelText('image-select')?.children.length !== 2) {
+    await new Promise(resolve => setTimeout(resolve, 100));
+  }
+  const select = screen.getByLabelText('image-select');
+  expect(select).toBeDefined();
+  expect(select.children.length).toEqual(2);
+
+  // Expect image:1 to be first since it's the last one in the history
+  expect(select.children[0].textContent).toEqual('image1:latest');
+  expect(select.children[1].textContent).toEqual('image2:latest');
+
+  // Expect input iso to be selected (it would have bg-purple-500 class)
+  const iso = screen.getByLabelText('iso-select');
+  expect(iso).toBeDefined();
+  expect(iso.classList.contains('bg-purple-500'));
+
+  // Expect input amd64 to be selected (it would have bg-purple-500 class)
+  const x86_64 = screen.getByLabelText('amd64-select');
+  expect(x86_64).toBeDefined();
+  expect(x86_64.classList.contains('bg-purple-500'));
+
+  // Expect input /tmp/image1 to be selected (it would have bg-purple-500 class)
+  const folder = screen.getByLabelText('folder-select');
+  expect(folder).toBeDefined();
+
+  // @ts-expect-error Ignore, as 'value' **does** exist in HTML element but is not recognized by TypeScript since it's using a custom svelte component
```

Comment:
I would not use ts-expect-error

you could try to check like isInstanceOf or something like that to see if there is a value field, etc